src/KlaviyoAPI.php

```diff
@@ -329,6 +331,25 @@ private function decodeJsonResponse( $response )
         return json_decode( '{}', true );
     }
 
+    /**
+     * Return jsone encoded rate limit array with details and the retryAfter value parsed.
+     * We build an easier object that tells you how long to retry after.
+     *
+     * @param mixed $response
+     * @return string
+     */
+    private function returnRateLimit ( $response )
+    {
+        $arr = explode(" ", $response['detail'] );
+        foreach ($arr as $value) {
+            if (intval($value) > 0) {
+                $response['retryAfter'] = intval($value);
```

Comment:
this really may not be the best way to do it, but here we are

Reply:
Two other possible methods but I don't necessarily think they are better:
- regex expression
- filter_var using FILTER_SANITIZE_NUMBER_FLOAT (but this would require some formatting if we always returned a float in the message)

Reply:
yea i was going to use this, but some fun stuff:

```php > var_dump(filter_var($s, FILTER_SANITIZE_NUMBER_FLOAT));
string(3) "123"
php > $s = 'test 123.00'
php > ;
php > var_dump(filter_var($s, FILTER_SANITIZE_NUMBER_FLOAT));
string(5) "12300"```